internal/s3manager/s3manager.go

```diff
@@ -0,0 +1,198 @@
+// Package s3manager implements the various functions needed to store and retrieve
+// files from an S3 API compatible endpoint (AWS S3, Minio, etc)
+package s3manager
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/epinio/epinio/helpers/kubernetes"
+	"github.com/google/uuid"
+	minio "github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"github.com/pkg/errors"
+	"gopkg.in/ini.v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Manager struct {
+	minioClient       *minio.Client
+	connectionDetails ConnectionDetails
+}
+
+type ConnectionDetails struct {
+	Endpoint string
+	// UseSSL Toggles the use of SSL for the s3 connection. If we deploy
+	// our own thing, linkderd will take care of this and we set useSSL to
+	// false.
+	// External S3 implementations should use SSL.
+	UseSSL          bool
+	AccessKeyID     string
+	SecretAccessKey string
+	Bucket          string
+	Location        string
+}
+
+func NewConnectionDetails(endpoint, key, secret, bucket, location string, useSSL bool) *ConnectionDetails {
+	return &ConnectionDetails{
+		Endpoint:        endpoint,
+		UseSSL:          useSSL,
+		AccessKeyID:     key,
+		SecretAccessKey: secret,
+		Bucket:          bucket,
+		Location:        location,
+	}
+}
+
+// Validate makes sure the S3 settings provided are valid
+// The user should either provide all the mandatory settings
```

Comment:
either ... or what if she doesn't ?
To make sure, the sentence is an incomplete `either/or` construct with the `or` branch missing.

Reply:
Fixed with: ```The user should provide all the mandatory settings or no settings at all```